### PR TITLE
Update STM32F1 conditions

### PR DESCRIPTION
**Pull request by mathiasvr**
### Description
Continuation of #20325.

The `ARDUINO_ARCH_STM32` flag was previously defined for all maple based STM32F1 environments.
This updates the code to mimic the current behavior even after removing `ARDUINO_ARCH_STM32`.

Clearly, not all these changes are necessary but included for completeness and discussion.

### Benefits
Removal of the `ARDUINO_ARCH_STM32` flag for STM32F1 maple environments.

### CODE REVIEW

File: Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h [hunk 1]
```diff
@@ -264,7 +264,7 @@
 
   // Remove compiler warning on an unused variable
   #ifndef UNUSED
-    #if defined(ARDUINO_ARCH_STM32) && !defined(STM32GENERIC)
+    #if ANY(ARDUINO_ARCH_STM32, ARDUINO_ARCH_STM32F1) && !defined(STM32GENERIC)
```
Comment (rhapsodyv) [inline Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h]: use `EITHER` when there're only 2 options.

File: Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h [hunk 2]
```diff
@@ -45,7 +45,7 @@
 //
 // Enable SD EEPROM to prevent infinite boot loop
 //
-#ifdef ARDUINO_ARCH_STM32
+#if ANY(ARDUINO_ARCH_STM32, ARDUINO_ARCH_STM32F1)
```
Comment (rhapsodyv) [inline Marlin/src/pins/stm32f1/pins_MKS_ROBIN.h]: use `EITHER` when there're only 2 options.

File: Marlin/src/core/serial.h [hunk 3]
```diff
@@ -85,7 +85,7 @@ extern uint8_t marlin_debug_flags;
 #define SERIAL_PRINTF(V...)     SERIAL_OUT(printf, V)
 #define SERIAL_FLUSH()          SERIAL_OUT(flush)
 
-#ifdef ARDUINO_ARCH_STM32
+#if ANY(ARDUINO_ARCH_STM32, ARDUINO_ARCH_STM32F1)
```
Comment (rhapsodyv) [inline Marlin/src/core/serial.h]: same

File: Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h (hunk 1 above)
Comment (mathiasvr) [reply Marlin/src/lcd/extui/lib/ftdi_eve_touch_ui/ftdi_eve_lib/compat.h]: Thanks!

**Comment by mathiasvr** [comment]
Closing for now as this will most likely be unnecessary.

**Pull request closed without merging**